nameable.rb

```diff
@@ -0,0 +1,17 @@
+class Nameable
+  def initialize(name)
+    @name = name
+  end
+
+  def correct_name
+    raise NotImplementedError, 'Subclasses should implement a correct_name method'
+  end
+end
+
+class Person < Nameable
+  attr_reader :name
+
+  def correct_name
+    name
+  end
+end
```

Comment:
I got this error while trying to run your code
![image](https://github.com/fpsapc/OOPSchoolLibrary/assets/78251927/903e1309-bc4b-4503-a59b-994ab44928ff)


Please fix the error and try the following code to decorate a class according to the [project requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md) 

```ruby
person = Person.new(22, 'maximilianus')
person.correct_name
capitalizedPerson = CapitalizeDecorator.new(person)
capitalizedPerson.correct_name
capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
capitalizedTrimmedPerson.correct_name
```